IntroducingMethods/src/cst1201/IntroducingMethods.java

```diff
@@ -3,33 +3,75 @@
 import java.util.Scanner;
 
 /**
- * TODO: Description of your class.
+ * Description of your class.
  *
- * @author TODO: Your name here.
+ * @author Francisco Latacela and Goutam Kundu .
  */
 public class IntroducingMethods {
 
     public static void main(String[] args) {
 
-        // TODO: Fill in the body with your code
+        //Fill in the body with your code
+        System.out.println("Please Enter a String: ");
+        Scanner sc = new Scanner(System.in);
+        getInputString(sc);
     }
 
-    // Given a Scanner, prompt the user for a String.  If the user enters an empty
-    // String, report an error message and ask for a non-empty String.  Return the
-    // String to the calling program.
+    /**
+     * Given a Scanner, prompts the user for a String. If the user enters an
+     * empty String, reports an error message and asks for a non-empty String.
+     * Returns the String to the calling program.
+     *
+     * @param inScanner The Scanner to use to prompt the user.
+     * @return The resulting String from the user.
+     */
     private static String getInputString(Scanner inScanner) {
-        // TODO: Fill in the body
-
-        // NOTE: Do not declare a Scanner in the body of this method.
+        //Fill in the body
+        String input = inScanner.nextLine();
+        while (input.length() == 0) {
+            System.out.println("You have entered an empty String, Please enter a non-empty String");
+            System.out.println("Please Enter a String: ");
+            input = inScanner.nextLine();
+        }
+        getWordCount(input);
+        return input;
     }
 
-    // Given a String return the number of words in the String.  A word is a sequence of 
-    // characters with no spaces.  Write this method so that the function call:
-    // 		int count = getWordCount("The quick brown fox jumped");
-    // results in count having a value of 5.  You will call this method from the main method.
-    // For this assignment you may assume that
-    // words will be separated by exactly one space.
+    /**
+     * Given a String, returns the number of words in the String. A word is a
+     * sequence of characters with no spaces. For example, the method call:
+     * <pre>
+     *      int count = getWordCount("The quick brown fox jumped");
+     * </pre> results in count having a value of 5. This method should be called
+     * from the main method. For this assignment, you may assume that words will
+     * be separated by exactly one space.
+     *
+     * @param input The String in question.
+     * @return The number of words in the String. Each word is separated by a
+     * single space.
+     */
     private static int getWordCount(String input) {
```

Comment:
:-1: Failed test case:
String input = "";
int expResult = 0;
actualResult: 1

The empty string should have no words.